### PR TITLE
Metrics e2e

### LINKED ISSUE #647: E2E test for Trigger metrics
## Expected Behavior
We need to have e2e tests for Trigger metrics using StackDriver as backend (event_count, dispatch_latencies, etc)

## Actual Behavior
We don't have 

## Additional Info
https://cloud.google.com/monitoring/api/metrics_knative

### LINKED ISSUE #648: E2E test for Broker metrics
## Expected Behavior
We need to have E2E tests for Broker metrics using StackDriver as backend (event_count, dispatch_latencies, etc) so that we do not regress

## Additional Info
https://cloud.google.com/monitoring/api/metrics_knative

### PR DESCRIPTION
**Pull request by ian-mi**
Fixes #647, fixes #648

## Proposed Changes

- Add TestGcpBrokerMetrics which tests that broker and trigger event count metrics are emitted.

### CODE REVIEW

File: test/e2e/test_gcp_broker.go [hunk 1]
```diff
@@ -52,17 +52,20 @@ GCPBrokerTestImpl tests the following scenario:
...
-		client.SetupStackDriverMetrics(t)
-	}
 	brokerURL, brokerName := createGCPBroker(client)
 	kngcphelpers.BrokerEventTransformationTestHelper(client, brokerURL, brokerName)
-	if assertMetrics {
-		lib.AssertBrokerMetrics(client)
-	}
+}
+
+func GCPBrokerMetricsTestImpl(t *testing.T, authConfig lib.AuthConfig) {
```
[the remark this answers is not in the document]
Comment (ian-mi) [reply test/e2e/test_gcp_broker.go:62]: I think it's better to keep tests focused on a single thing. In this case it makes it easier to tell when a test is failing or flaky due to a broker issue or a metrics specific issue. I haven't yet had a chance to look at refactoring the pubsub and storage tests.

**Comment by zargarpur** [comment]
Nice!
/lgtm

**Pull request merged**